File: aoc1701.cc
#include "iostream"

using	namespace std;

int	main()
{
	string		s;
	int		i;
	int		end, tt = 0;
	int		half, t2 = 0;

	cin >> s;
	end = s.length() - 1;
	i = 0;
	while (1)
	{
		if (i == end && s[0] == s[0])
		{
			tt += s[i] - '0';
			break ;
		}
		if (i ^ end && s[i + 1] == s[i])
		{
			tt += s[i] - '0';
		}
		++i;
	}
	i = -1;
	half = s.length() / 2;
	while (++i < half)
	{
		if (i + half <= end && s[i] == s[i + half])
		{
			t2 += (s[i] - '0') * 2;
		}
	}
	cout << "Star 1: " << tt << endl;
	cout << "Star 2: " << t2 << endl;
}

File: aoc1701.py
#* ******************************************** *#
#*                                              *#
#*              \\             /`/``            *#
#*              ~\o o_       0 0\               *#
#*              / \__)      (u  ); _  _         *#
#*       / \/ \/  /           \  \/ \/ \        *#
#*      /(   . . )            (         )\      *#
#*     /  \_____/              \_______/  \     *#
#*         []  []               [[] [[]    *.   *#
#*         []] []]              [[] [[]         *#
#*                                              *#
#* ************************************ nuo *** *#

s = input()
i = 0
p1 = 0
p2 = 0
while i < len(s):
    if i > 0:
        if i == len(s) - 1 and s[i] == s[-1]:
            p1 += int(s[-1])
        if s[i] == s[i - 1]:
            p1 += int(s[i])
    half = i+len(s)//2 if i<len(s)//2 else i-len(s)//2
    if s[i] == s[half]:
        p2 += int(s[i])
    i += 1
print(p1)
print(p2)
